airflow/dag_processing/bundles/git.py

```diff
@@ -66,20 +83,37 @@ def __init__(self, *, repo_url: str, tracking_ref: str, subdir: str | None = Non
             self.repo.head.set_reference(self.repo.commit(self.version))
             self.repo.head.reset(index=True, working_tree=True)
         else:
-            self.refresh()
+            self._refresh()
+
+    def _ssh_hook(self):
+        try:
+            from airflow.providers.ssh.hooks.ssh import SSHHook
+        except ImportError as e:
+            raise AirflowOptionalProviderFeatureException(e)
+        return SSHHook(**self.ssh_conn_kwargs)
+
+    def initialize(self) -> None:
+        if self.ssh_conn_kwargs:
+            if not self.repo_url.startswith("git@") and not self.repo_url.endswith(".git"):
```

Comment:
```suggestion
            if not self.repo_url.startswith("git@") or not self.repo_url.endswith(".git"):
```

We need to check that they are there individually.